keras_hub/src/models/gemma3/gemma3_utils.py

```diff
@@ -0,0 +1,255 @@
+from enum import Enum
+from typing import Optional, Union, Tuple, Dict, List
+import tensorflow as tf
+from keras import ops
+import logging
+import numpy as np
+
+import math
+import itertools
+import re
+
+import PIL.Image
+import PIL.ImageOps
+
+logger = logging.getLogger(__name__)
+
+class ExplicitEnum(str, Enum):
+    """
+    Enum with more explicit error message for missing values.
+    """
+
+    @classmethod
+    def _missing_(cls, value):
+        raise ValueError(
+            f"{value} is not a valid {cls.__name__}, please select one of {list(cls._value2member_map_.keys())}"
+        )
+
+class ChannelDimension(ExplicitEnum):
+    FIRST = "channels_first"
+    LAST = "channels_last"
+
+def infer_channel_dimension_format(
+    image: np.ndarray, num_channels: Optional[Union[int, Tuple[int, ...]]] = None
+) -> ChannelDimension:
+    """
+    Infers the channel dimension format of `image`.
+
+    Args:
+        image (`np.ndarray`):
+            The image to infer the channel dimension of.
+        num_channels (`int` or `Tuple[int, ...]`, *optional*, defaults to `(1, 3)`):
```

Comment:
usual convention for args is
num_channels: int or tuple of 2 integers,  optional. The number of channels of the image. Defaults to `(1, 3)`.

You can apply the same format for all the Args details in this PR.